tests/backup/backup_share_test.go

```diff
@@ -4215,3 +4215,310 @@ var _ = Describe("{SwapShareBackup}", func() {
+			})
+		}
+		ctx, err := backup.GetAdminCtxFromSecret()
+		log.FailOnError(err, "Fetching px-central-admin ctx")
+		cleanupUserObjectsFromAdmin := func(user string) {
+			defer GinkgoRecover()
+			Step(fmt.Sprintf("Delete user %s schedule backups, backup schedule and schedule policy from the admin", user), func() {
+				log.InfoD(fmt.Sprintf("Deleting user %s schedule backups, backup schedule and schedule policy from the admin", user))
+				nonAdminCtx, err := backup.GetNonAdminCtx(user, commonPassword)
+				log.FailOnError(err, "failed to fetch user %s ctx", user)
+				allScheduleBackupNames, err := Inst().Backup.GetAllScheduleBackupNames(nonAdminCtx, userScheduleNameMap[user], orgID)
+				log.FailOnError(err, "failed to get all schedule backup names with schedule name %s of the user %s", userScheduleNameMap[user], user)
+				for i := len(allScheduleBackupNames) - 1; i >= 0; i-- {
+					backupName := allScheduleBackupNames[i]
+					backupUid, err := Inst().Backup.GetBackupUID(nonAdminCtx, backupName, orgID)
+					log.FailOnError(err, "failed to fetch backup %s uid of the user %s", backupName, user)
+					_, err = DeleteBackupWithClusterUID(backupName, backupUid, SourceClusterName, orgID, ctx)
+					log.FailOnError(err, "failed to delete schedule backup %s of the user %s", backupName, user)
+				}
+				scheduleUid, err := Inst().Backup.GetBackupScheduleUID(nonAdminCtx, userScheduleNameMap[user], orgID)
+				log.FailOnError(err, "failed to fetch backup schedule %s uid of the user %s", userScheduleNameMap[user], user)
+				err = DeleteScheduleWithUID(userScheduleNameMap[user], scheduleUid, orgID, ctx)
+				log.FailOnError(err, "failed to delete schedule %s of the user %s", userScheduleNameMap[user], user)
+			})
+			Step(fmt.Sprintf("Delete user %s backups from the admin", user), func() {
+				log.InfoD(fmt.Sprintf("Deleting user %s backups from the admin", user))
+				nonAdminCtx, err := backup.GetNonAdminCtx(user, commonPassword)
+				log.FailOnError(err, "failed to fetch user %s ctx", user)
+				for backupName := range userBackupMap[user] {
+					backupUid, err := Inst().Backup.GetBackupUID(nonAdminCtx, backupName, orgID)
+					log.FailOnError(err, "failed to fetch backup %s uid of the user %s", backupName, user)
+					_, err = DeleteBackupWithClusterUID(backupName, backupUid, userClusterMap[user][SourceClusterName], orgID, ctx)
+					log.FailOnError(err, "failed to delete backup %s of the user %s", backupName, user)
+				}
+			})
+			Step(fmt.Sprintf("Delete user %s restores from the admin", user), func() {
+				log.InfoD(fmt.Sprintf("Deleting user %s restores from the admin", user))
+				for restoreUid, restoreName := range userRestoreMap[user] {
+					err = DeleteRestoreWithUID(restoreName, restoreUid, orgID, ctx)
+					log.FailOnError(err, "failed to delete restore %s of the user %s", restoreName, user)
+				}
+			})
+			Step(fmt.Sprintf("Wait for the backups and backup schedule to be deleted"), func() {
```

Comment:
can we make use of DeleteBackupAndWait and DeleteScheduleWithUIDandWait with delete steps here instead of a separate step.

Reply:
DeleteBackupAndWait is an old function. For DeleteScheduleWithUIDandWait, I am deleting the schedule using the admin ctx and waiting for it to be deleted from the non-admin ctx.